lib/ansible/module_utils/cloudfront.py

```diff
@@ -0,0 +1,122 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) 2017 Willem van Ketwich
+#
+# This module is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This software is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this software.  If not, see <http://www.gnu.org/licenses/>.
+#
+# Author:
+#   - Willem van Ketwich <willem@vanketwich.com.au>
+#
+# Common functionality to be used by the modules:
+#   - cloudfront_distribution
+#   - cloudfront_invalidation
+#   - cloudfront_origin_access_identity
+
+
+"""
+Miscellaneous helpers for processing cloudfront data
+"""
+
+
+def change_dict_key_name(dictionary, old_key, new_key):
+    if old_key in dictionary:
+        dictionary[new_key] = dictionary.get(old_key)
+        dictionary.pop(old_key, None)
+    return dictionary
+
+
+def snake_dict_to_pascal_dict(snake_dict):
```

Comment:
`snake_dict_to_camel_dict` is different from `snake_dict_to_pascal_dict` in that pascal has the first letter capitalised, whereas camel has first letter lower-cased. This is important as this function is used for calling boto with parameters that are case-sensitive. If it were just the first-level items of the dict, it could use the snake_dict_to_camel_dict function with .capitalize() but there are objects that have nested object that have nested objects... and so on.